lib/class/api.class.php

```diff
@@ -251,25 +264,49 @@ public static function handshake($input)
                 $db_results = Dba::read($sql);
                 $catalog    = Dba::fetch_assoc($db_results);
 
-                echo XML_Data::keyed_array(array('auth'=>$token,
-                    'api'=>self::$version,
-                    'session_expire'=>date("c",time()+AmpConfig::get('session_length')-60),
-                    'update'=>date("c",$row['update']),
-                    'add'=>date("c",$row['add']),
-                    'clean'=>date("c",$row['clean']),
-                    'songs'=>$song['song'],
-                    'albums'=>$album['album'],
-                    'artists'=>$artist['artist'],
-                    'playlists'=>$playlist['playlist'],
-                    'videos'=>$vcounts['video'],
-                    'catalogs'=>$catalog['catalog']));
+                if ($outputFormat == 'json') {
+                  echo json_encode(
+                      array(
```

Comment:
In truth, I got a little frustrated and made a cheap fix until I properly replace the XML function.

But your right I should make a variable and just store the array and use that, smarter than what I did.